presto-main/src/main/java/io/prestosql/execution/SqlQueryManager.java

```diff
@@ -313,4 +326,24 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce max raw data size at the query level
+     */
+    public void enforceQueryMaxInputDataSizeLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            Optional<DataSize> limitOpt = Optional.ofNullable(getQueryMaxDataSize(query.getSession())
+                    .flatMap(sessionLimit -> maxQueryInputDataSize.map(serverLimit -> Ordering.natural().min(serverLimit, sessionLimit)))
+                    .orElse(maxQueryInputDataSize.orElse(null)));
```

Comment:
Now we have to moved to Java 11 so I think we can use Optional#or(() -> maxQueryInputDataSize) which could simplify the above statement